pkg/appolly/discover/attacher.go

```diff
@@ -273,6 +278,21 @@ func (ta *traceAttacher) harvestRoutes(ie *ebpf.Instrumentable, reused bool) {
 	}
 }
 
+func (ta *traceAttacher) harvestRoutes(ie *ebpf.Instrumentable, reused bool) {
+	if delay, delayTime := ta.routeHarvester.HarvestRoutesDelay(ie.FileInfo); delay {
+		procAge := ta.processAgeFunc(ie.FileInfo.Pid)
+		if procAge < delayTime {
+			time.AfterFunc(delayTime-procAge, func() {
+				ta.harvestRoutesProcessor(ie, reused)
```

Comment:
What would happen if the target java process, during/after the timeout:

- the target process died
- the target pid was reused for a different process

Maybe a cancellable context could be used, or pid re-validation/age matching could happen - to ensure we're still acting on the same process?